### PR TITLE
Add function for single page of items

### LINKED ISSUE #6: Make the paginated entries public
In the context of Read It Later; the get_entries in some cases could lead to loading a thousands of hundreds of articles. In order to avoid making the app unusable for a few minutes my plan is to only load a few articles and load  the rest in the background and load them only if they are requested (on scroll).

Allowing to fetch paginated entries makes completely sense to have here to benefit other clients

### PR DESCRIPTION
**Pull request by swalladge**
Fixes #6 

This adds a function to retrieve a single page of results. `get_entries` and `get_entries_with_filter` automatically resolve pagination to return all entries. Adding this allows an option for the client to have more control over number of network requests for various purposes. See #6 for use case.

This probably needs some more testing before merging...

### CODE REVIEW

**Comment by swalladge** [comment]
@bilelmoussaoui would you mind checking this out to see if it both works and suits your use case? :)

**Comment by bilelmoussaoui** [comment]
I will take a look later this evening, thanks a lot!

File: src/client.rs [hunk 1]
```diff
@@ -410,24 +450,47 @@ impl Client {
...
+        let params = RequestEntriesFilter { page, filter };
+        let json: PaginatedEntries = self
+            .smart_json_q(Method::GET, EndPoint::Entries, &params, UNIT)
+            .await?;
+
+        Ok(Page {
+            limit: json.limit,
+            page: json.page,
+            pages: json.pages,
+            total: json.pages,
```
[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/client.rs]: total seems to return the total of articles and pages the total number of pages

[the remark this answers is not in the document]
Comment (swalladge) [reply src/client.rs]: Cool, I also had a play around and updated the naming of everything to be more clear.

**Pull request merged**